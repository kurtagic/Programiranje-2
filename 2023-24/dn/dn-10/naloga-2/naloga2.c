
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

Tocka projekcija(Tocka t, Premica p) {
    // popravite / dopolnite ...
    return (Tocka) { (Ulomek) {0, 1}, (Ulomek) {0, 1} };
}

#ifndef test

int main() {
    return 0;
}

#endif
