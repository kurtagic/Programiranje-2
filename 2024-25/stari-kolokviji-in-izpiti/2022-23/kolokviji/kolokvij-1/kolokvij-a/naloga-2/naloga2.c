
/*
 * Prevajanje in zagon testnega programa testXY.c:
 *
 * gcc -D=test testXY.c naloga2.c
 * ./a.out
 *
 * Zagon testne skripte ("sele potem, ko ste prepri"cani, da program deluje!):
 *
 * export name=naloga2
 * make test
 *
 * Javni testni primeri (po te"zavnosti):
 * -- 02, 03, 04: en sam vhodni niz;
 * -- 05, 06, 07: vsi vhodni nizi so enako dolgi;
 * -- 01, 08, 09, 10: splo"sni primeri.
 *
 * Javni testni primeri (po na"cinu priprave):
 * -- 01: primer iz besedila;
 * -- 01, 02, 05: ro"cno ustvarjeni;
 * -- ostali: samodejno generirani.
 */

#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <string.h>

// po potrebi dopolnite ...

//=============================================================================

char** poStolpcih(char** nizi, int stVhodnih, int* stIzhodnih) {
    // popravite / dopolnite ...
    return NULL;
}

//=============================================================================

#ifndef test

int main() {
    // "Ce "zelite funkcijo poStolpcih testirati brez testnih primerov,
    // dopolnite to funkcijo in prevedite datoteko na obi"cajen na"cin
    // (gcc naloga2.c).
    return 0;
}

#endif
