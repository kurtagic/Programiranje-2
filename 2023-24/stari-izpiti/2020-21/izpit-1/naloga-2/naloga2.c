#include <stdio.h>
#include <stdlib.h>

#include "naloga2.h"

Vozlisce *sticisce(Vozlisce *a, Vozlisce *b)
{
    Vozlisce *current_a = a;

    while (current_a != NULL)
    {
        Vozlisce *current_b = b;
        while (current_b != NULL)
        {
            if (current_a == current_b)
            {
                return current_a;
            }

            current_b = current_b->n;
        }

        current_a = current_a->n;
    }

    return NULL;
}

// TESTS
Vozlisce *naredi(int vred)
{
    Vozlisce *novo = malloc(sizeof(Vozlisce));
    novo->vrednost = vred;
    return novo;
}

int main()
{
    Vozlisce *prvo = naredi(1);
    Vozlisce *drugo = naredi(2);
    Vozlisce *tretje = naredi(3);
    Vozlisce *skupno = naredi(10);

    Vozlisce *cetrto = naredi(4);
    Vozlisce *peto = naredi(5);

    prvo->n = drugo;
    drugo->n = tretje;
    tretje->n = skupno;
    cetrto->n = skupno;

    printf("%d\n", sticisce(prvo, cetrto)->vrednost);

    return 0;
}
