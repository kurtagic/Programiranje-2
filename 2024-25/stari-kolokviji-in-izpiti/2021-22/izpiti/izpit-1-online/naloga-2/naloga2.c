
/*
Ro"cno poganjanje testnih programov (npr. test01.c):

gcc -Dtest test01.c naloga2.c
./a.out

Samodejno testiranje:

export name=naloga2
make test

Testni primeri:

test01: primer iz besedila
test02..test08: ro"cno izdelani kratki testi
test09..test10: samodejno izdelani, zadnji element prvega seznama < prvi element drugega seznama
test11..test16: samodejno izdelani, splo"sni

.c: testni program (prebere testne podatke, pokli"ce va"so funkcijo in izpi"se rezultat)
.dat: testni podatki
.out: pri"cakovani izhod
.res: izhod va"sega programa (pri poganjanju z make)
*/

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

// Lahko dodate "se kak #include, pomo"zno funkcijo ipd.

//============================================================================

Vozlisce* zlij(Vozlisce* a, Vozlisce* b) {
    // dopolnite ...

    return NULL;   // popravite
}

//============================================================================

// Vrstici z #ifndef in #endif pustite pri miru!

#ifndef test

int main() {
    return 0;
}

#endif
