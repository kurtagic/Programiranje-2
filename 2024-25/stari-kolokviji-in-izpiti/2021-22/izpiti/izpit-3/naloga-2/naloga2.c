
/*
Ro"cno poganjanje testnih programov (npr. test01.c):

gcc -Dtest test01.c naloga2.c
./a.out

Samodejno testiranje:

export name=naloga2
make test

Testni primeri:

01: primer iz besedila
02..04: ena sama vrstica
05..06: vsaka neprazna vrstica je naslovna
07..10: splo"sni primeri

Datoteke:

test*.c: testni program (prebere testne podatke, pokli"ce va"so funkcijo in izpi"se rezultat)
test*.dat: testni podatki
test*.out: pri"cakovani izhod testnega programa
test*.res: dejanski izhod testnega programa (pri poganjanju z make)
*/

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

// Lahko dodate "se kak #include, pomo"zno funkcijo ipd.

//============================================================================

void vstaviH1(Vozlisce* zacetek) {
    // dopolnite ...
}

//============================================================================

// Vrstici z #ifndef in #endif pustite pri miru!

#ifndef test

int main() {
    // Tole dopolnite, "ce "zelite funkcijo lastnoro"cno preveriti.
    // V tem primeru program po"zenite preprosto kot
    // gcc naloga3.c
    // ./a.out
    return 0;
}

#endif
