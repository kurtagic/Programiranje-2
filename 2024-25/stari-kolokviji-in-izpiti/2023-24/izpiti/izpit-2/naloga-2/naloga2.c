
/*
 * Prevajanje in zagon testnega programa testXY.c:
 *
 * gcc -Dtest testXY.c naloga2.c
 * ./a.out
 *
 * Zagon testne skripte ("sele potem, ko ste prepri"cani, da program deluje!):
 *
 * export name=naloga2
 * make test
 *
 * Javni testni primeri:
 *
 * 02..03: zaporedje podatkov a_1, a_1, a_2, a_2, a_3, a_3, ... (a_i != a_{i+1})
 * 04..05: nikoli nimamo ve"c kot dveh zaporednih vozli"s"c z enako vrednostjo
 *         komponente <podatek>
 * 01, 06..10: splo"sni primeri
 */

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

//=============================================================================

// po potrebi dopolnite ...

void obdelaj(Vozlisce* zacetek) {
    // dopolnite ...
}

#ifndef test

int main() {
    // "Ce "zelite funkcijo <obdelaj> testirati brez testnih primerov,
    // dopolnite to funkcijo in prevedite datoteko na obi"cajen na"cin 
    // (gcc naloga2.c).
    return 0;
}

#endif
