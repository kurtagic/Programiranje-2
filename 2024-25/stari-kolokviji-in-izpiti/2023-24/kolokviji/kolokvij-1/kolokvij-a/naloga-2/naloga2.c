#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

void racionaliziraj(char** nizi)
{
	// poglej vse kombinacije parov
	for (int i = 0; nizi[i] != NULL; i++)
	{
		for (int j = 0; nizi[j] != NULL; j++)
		{
			if (i == j)
			{
				continue;
			}

			// ce sta niza enaka, spremen j-ti pointer da kaze na i-ti pointer (vedno j > i)
			if (!strcmp(nizi[i], nizi[j]))
			{
				nizi[j] = nizi[i];
			}
		}
	}
}

#ifndef test
int main() { return 0; }
#endif
