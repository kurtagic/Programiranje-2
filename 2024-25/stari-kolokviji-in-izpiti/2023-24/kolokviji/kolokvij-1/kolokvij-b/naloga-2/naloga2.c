
/*
 * Prevajanje in zagon testnega programa testXY.c:
 *
 * gcc -Dtest testXY.c naloga2.c
 * ./a.out
 *
 * Zagon testne skripte ("sele potem, ko ste prepri"cani, da program deluje!):
 *
 * export name=naloga2
 * make test
 */

#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

//=============================================================================

// po potrebi dopolnite ...

void razbohoti(char** nizi) {
    // dopolnite ...
}

//=============================================================================

#ifndef test

int main() {
    // "Ce "zelite funkcijo <razbohoti> testirati brez testnih primerov,
    // dopolnite to funkcijo in prevedite datoteko na obi"cajen na"cin 
    // (gcc naloga2.c).
    return 0;
}

#endif
