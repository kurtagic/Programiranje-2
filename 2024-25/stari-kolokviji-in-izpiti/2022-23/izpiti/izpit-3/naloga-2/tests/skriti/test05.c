
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>

#include "naloga2.h"

int* SLIKA[] = {
    (int[]) {33, 7, 30, 21, 32, 0, 7, 0, 48, 0, 86, 28, 1, 0, 25, 0, 1, 0, 0, 22, 0, 4, 32, 3, 0, 56, 0, 0, 74, 60, 74, 0, 16, 90, 11, 36, 0, 0, 38, 66, 83, 49, 50, 8, 0, 74, 0, 0, 26, 0, 0, 0, 44, 0, 49, 98, 77, 0, 79, 0, 0, 87, 0, 0, 0, 0, 0, 0, 79, 0, 100, 3, 37, 0, 0, 0, 79, 0, 0, 0, 96, 66, 0, 0, 0, 0, 0, 28, 0, 37, 62, 103, 0, 0, 109, 12, 0, 108, 192, 143, 79, 253, 211, 196, 234, 426, 296, 406, 536, 648, 626, 675, 747, 778, 1021, 970, 1231, 1175, 1277, 1490, 1543, 1642, 1642, 1709, 1881, 1813, 1824, 1860, 2023, 2023, 2023, 2023, 1912, 2023, 1845, 1765, 1806, 1794, 1709, 1525, 1564, 1407, 1306, 1197, 1216, 1057, 922, 890, 803, 625, 734, 501, 448, 507, 479, 333, 253, 207, 112, 248, 91, 34, 196, 40},
    (int[]) {0, 21, 15, 8, 19, 96, 0, 92, 89, 74, 0, 0, 0, 0, 71, 67, 90, 58, 91, 19, 0, 99, 0, 0, 0, 0, 57, 25, 0, 97, 52, 70, 0, 0, 0, 5, 23, 0, 92, 0, 0, 97, 96, 49, 17, 0, 0, 0, 25, 0, 81, 64, 0, 58, 0, 0, 83, 100, 0, 46, 81, 23, 23, 0, 0, 0, 79, 0, 0, 55, 93, 0, 0, 84, 0, 0, 0, 40, 74, 0, 95, 0, 0, 0, 0, 0, 92, 23, 0, 0, 49, 88, 72, 91, 62, 140, 20, 145, 84, 135, 200, 98, 117, 305, 382, 391, 296, 405, 591, 518, 633, 624, 806, 976, 1060, 1114, 1151, 1330, 1361, 1521, 1476, 1636, 1730, 1671, 1853, 1805, 1896, 1893, 1960, 1995, 1983, 2023, 2023, 1964, 1854, 1911, 1765, 1678, 1684, 1563, 1452, 1454, 1423, 1251, 1187, 1025, 976, 954, 798, 749, 602, 480, 428, 373, 317, 356, 202, 220, 134, 182, 93, 118, 37, 9},
    (int[]) {8, 40, 87, 0, 1, 0, 0, 11, 78, 21, 66, 0, 74, 0, 0, 98, 90, 63, 0, 71, 10, 0, 0, 64, 0, 58, 98, 0, 0, 0, 0, 40, 50, 0, 0, 65, 0, 0, 0, 92, 0, 85, 0, 21, 0, 40, 0, 0, 0, 0, 0, 39, 0, 17, 81, 0, 0, 73, 22, 71, 0, 23, 19, 66, 39, 0, 0, 0, 57, 24, 0, 81, 19, 29, 0, 5, 0, 0, 0, 30, 0, 0, 68, 0, 68, 0, 52, 3, 15, 0, 0, 0, 37, 98, 72, 126, 72, 18, 31, 95, 214, 273, 124, 242, 193, 249, 448, 429, 581, 507, 739, 633, 729, 980, 989, 1143, 1127, 1295, 1296, 1411, 1508, 1612, 1648, 1788, 1816, 1814, 1856, 1906, 2020, 2023, 1975, 1933, 1975, 2009, 2023, 1889, 1738, 1836, 1663, 1656, 1601, 1458, 1274, 1342, 1229, 1120, 1045, 867, 763, 773, 597, 516, 579, 438, 473, 330, 225, 349, 169, 220, 69, 33, 167, 60},
};

int N = 3;
int M = 164;
int D = 1;

int main() {
    int** nova = zmehcaj(SLIKA, N, M, D);
    int noviN = N - 2 * D;
    int noviM = M - 2 * D;

    for (int i = 0; i < noviN; i++) {
        for (int j = 0; j < noviM; j++) {
            printf("%5d", nova[i][j]);
        }
        printf("\n");
    }

    for (int i = 0; i < noviN; i++) {
        free(nova[i]);
    }
    free(nova);

    return 0;
}
