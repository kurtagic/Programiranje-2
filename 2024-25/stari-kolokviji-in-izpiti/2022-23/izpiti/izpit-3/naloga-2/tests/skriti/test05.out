   26   25   23   19   17   22   36   45   51   30   28   11   19   29   49   51   43   36   23   25   16   22   11   20   29   32   28   28   39   43   33   29   18   23   15   14   17   32   41   52   51   50   26   23   14   12    5    5   14   20   25   24   27   33   43   47   48   43   33   36   28   26   16   11   13    8   23   23   45   39   37   28   18   13    9   13   21   16   32   31   36   14   15    7   23   22   23   11   18   37   45   54   60   78   68   71   74  105  130  165  174  202  222  290  323  371  443  513  597  622  690  772  898 1003 1089 1170 1249 1350 1431 1537 1604 1675 1748 1794 1839 1851 1915 1967 2002 2000 1985 1987 1958 1920 1844 1798 1741 1678 1601 1520 1437 1345 1269 1180 1086  984  890  802  716  619  542  474  449  400  329  275  207  204  146  120   93   77
